app/models/course_user.rb

```diff
@@ -1,5 +1,33 @@
 class CourseUser < ActiveRecord::Base
+  include Workflow
+  workflow do
+    state :pending do
+      event :approve, transitions_to: :approved
+      event :reject, transitions_to: :rejected
+    end
+    state :approved
+    state :rejected
+  end
+
+  stampable
   belongs_to :user, inverse_of: :course_users
   belongs_to :course, inverse_of: :course_users
+
   enum role: { student: 0, teaching_assistant: 1, manager: 2, owner: 3 }
+
+  scope :staff, -> { where(role: staff_roles) }
+
+  # Returns staff roles
+  #
+  # @return [Array] array of staff role enums
+  def self.staff_roles
+    [roles[:teaching_assistant], roles[:manager], roles[:owner]]
+  end
+
+  # Test whether this course_user is a staff (i.e. teaching_assistant, manager or owner)
+  #
+  # @return [Boolean] True if course_user is a staff
+  def staff?
+    self.class.staff_roles.include?(self[:role])
```

Comment:
`CourseUser.STAFF_ROLES...`

How come this is called CourseUser and not Course::User :confused: